MapboxNavigation/RouteVoiceController.swift

```diff
@@ -193,15 +238,29 @@ open class RouteVoiceController: NSObject, AVSpeechSynthesizerDelegate, AVAudioP
         legProgress = routeProgress.currentLegProgress
         guard let instruction = routeProgress.currentLegProgress.currentStepProgress.currentSpokenInstruction else { return }
         lastSpokenInstruction = instruction
-        speak(instruction)
+        
+        for (stepIndex, step) in routeProgress.currentLegProgress.leg.steps.suffix(from: routeProgress.currentLegProgress.stepIndex).enumerated() {
+            let adjustedStepIndex = stepIndex + routeProgress.currentLegProgress.stepIndex
+            
+            guard adjustedStepIndex < routeProgress.currentLegProgress.stepIndex + stepsAheadToCache else { continue }
+            guard let instructions = step.instructionsSpokenAlongStep else { continue }
+            
+            for instruction in instructions {
+                guard spokenInstructionsForRoute.object(forKey: instruction.ssmlText as NSString) == nil else { continue }
+                
+                cacheAndPlay(instruction: instruction, play: false)
```

Comment:
The new code targeting the Mapbox Speech API should live in its own RouteVoiceController subclass for the following reasons:

* It keeps the failable network request and precaching code more modular.
* It makes the fallback to AVSpeechSynthesizer more clear.
* Even if the Mapbox Speech API is available out of the box, a developer may already be using a third-party speech recognition SDK that comes with its own speech synthesizer. Or they may be using a third-party speech synthesizer custom-tailored to a language that the Speech API doesn’t support.

We can have the navigation SDK use this new voice controller by default, but it should still be possible to swap in a different voice controller.